src/API/Endpoints/Reports/DonationsVsIncome.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+/**
+ * Reports base endpoint
+ *
+ * @package Give
+ */
+
+namespace Give\API\Endpoints\Reports;
+
+class DonationsVsIncome extends Endpoint {
+
+	public function __construct() {
+		$this->endpoint = 'donations-vs-income';
+	}
+
+	public function get_report($request) {
+
+		$start = date_create($request['start']);
+		$end = date_create($request['end']);
+		$diff = date_diff($start, $end);
+
+		$data = [];
+
+		switch(true) {
+			case ($diff->days > 900):
+				$data = $this->get_data($start, $end, 'P1Y', 'Y');
+				break;
+			case ($diff->days > 700):
+				$data = $this->get_data($start, $end, 'P6M', 'F Y');
+				break;
+			case ($diff->days > 400):
+				$data = $this->get_data($start, $end, 'P3M', 'F Y');
+				break;
+			case ($diff->days > 120):
+				$data = $this->get_data($start, $end, 'P1M', 'M Y');
+				break;
+			case ($diff->days > 30):
+				$data = $this->get_data($start, $end, 'P7D', 'M jS');
+				break;
+			case ($diff->days > 10):
+				$data = $this->get_data($start, $end, 'P3D', 'M jS');
+				break;
+			case ($diff->days > 4):
+				$data = $this->get_data($start, $end, 'P1D', 'l');
+				break;
+			case ($diff->days > 1):
+				$data = $this->get_data($start, $end, 'PT6H', 'D ga');
+				break;
+			case ($diff->days >= 0):
+				$data = $this->get_data($start, $end, 'PT1H', 'D ga');
+				break;
+		}
+
+		// Add caching logic here...
+
```

Comment:
@henryholtgeerts unnecessary line

Reply:
Removed this comment! ref: https://github.com/impress-org/givewp/pull/4412/commits/e0254ef763680698d0cf1544c07ec6c92b9ec220